octoprint_mrbeam/static/js/working_area.js

```diff
@@ -2490,80 +2503,101 @@ $(function () {
+            compSvg.selectAll(".deleteBeforeRendering").remove();
+            const targetDefs = compSvg.select("svg>defs");
 
-            // remove all items maked with deleteBeforeRendering class
-            var dels = compSvg.selectAll(".deleteBeforeRendering");
-            if (dels && dels.length > 0) {
-                for (var i = 0; i < dels.length; i++) {
-                    dels[i].remove();
-                }
-            }
-            // TODO why not shorter?
-            // compSvg.selectAll('.deleteBeforeRendering').remove();
-
-            // embed the fonts as dataUris
+            // if text in document
             if (userContent.selectAll(".userText").length > 0) {
+                // embed the fonts as dataUris
                 $("#compSvg defs").append(
                     '<style id="quickTextFontPlaceholder" class="quickTextFontPlaceholder deleteAfterRendering"></style>'
                 );
                 self._qt_copyFontsToSvg(
                     compSvg.select(".quickTextFontPlaceholder").node
                 );
-            }
 
-            self.renderInfill(
-                compSvg,
-                namespaces,
-                wPT,
-                hPT,
-                fillAreas,
-                engraveStroke,
-                wMM,
-                hMM,
-                pxPerMM,
-                function (svgWithRenderedInfill) {
-                    callback(
-                        self._wrapInSvgAndScale(
-                            svgWithRenderedInfill,
-                            namespaces
-                        )
-                    );
-                    $("#compSvg").remove();
+                // copy curved textPaths to defs
+                const allTextPaths = snap.selectAll(
+                    "defs>.quicktext_curve_path"
+                );
+                for (let i = 0; i < allTextPaths.length; i++) {
+                    const tp = allTextPaths[i];
+                    const original_id = tp.attr("id");
+                    const clone = tp.clone();
+                    const destTextPath = clone.appendTo(targetDefs);
+                    // restore id to keep references working
+                    destTextPath.attr({
+                        id: original_id,
+                        "mb:id": original_id,
+                    });
                 }
-            );
+            }
+
+            // embed filters
+            // copy defs for filters (e.g. imgCropping, imgSharpening, etc...)
+            var originalFilters = snap.selectAll("defs>filter");
+            for (let i = 0; i < originalFilters.length; i++) {
+                const original_id = originalFilters[i].attr("id");
+                const clone = originalFilters[i].clone();
+                const destFilter = clone.appendTo(targetDefs);
+                // restore id to keep references working
+                destFilter.attr({ id: original_id, "mb:id": original_id });
+            }
```

Comment:
what is this for? It only seems to modify a clone and then the clone is not used

Reply:
You need to learn more about Snapsvg. A clone is directly hooked in the DOM. No need to "use" it.